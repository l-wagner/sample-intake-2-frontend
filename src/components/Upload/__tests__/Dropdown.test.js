import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import TestDropdown from '../Dropdown'

const Dropdown = props => <TestDropdown {...props} />

describe('Render Dropdown', () => {
  it('render Dropdown correctly', () => {
    const DropdownComponent = renderer.create(<Dropdown />).toJSON()
    expect(DropdownComponent).toMatchSnapshot()
  })

  it('render Dropdown correctly with loading animation', () => {
    const wrapper = mount(<Dropdown dynamic loading={true} />)
    const progress = wrapper.find('LinearProgress').first()
    expect(progress.exists()).toBe(true)
  })
})
 