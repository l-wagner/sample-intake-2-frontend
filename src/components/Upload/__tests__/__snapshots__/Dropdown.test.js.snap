// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Dropdown render Dropdown correctly 1`] = `
<div
  className="Dropdown-textField-1"
>
  <div
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-labelledby="downshift-0-label"
    aria-owns={null}
    role="combobox"
  >
    <div
      className="MuiFormControl-root-2 MuiFormControl-fullWidth-5"
    >
      <div
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-labelledby="downshift-0-label"
        className="MuiInputBase-root-19 MuiInput-root-6 MuiInput-underline-10 MuiInputBase-formControl-20 MuiInput-formControl-7 MuiInputBase-adornedEnd-24"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="off"
          className="MuiInputBase-input-29 MuiInput-input-14 MuiInputBase-inputAdornedEnd-35"
          disabled={false}
          id="downshift-0-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          required={false}
          type="text"
          value=""
        />
        <div
          className="MuiInputAdornment-root-36 MuiInputAdornment-positionEnd-39"
        >
          <button
            aria-label="Toggle menu open"
            className="MuiButtonBase-root-47 MuiIconButton-root-41"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-46"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-50"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-60"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
